File: java/P003.java
/*
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600851475143 ?
*/

package from_01_to_10;

public class Problem_03 {
	public static void main(String[] args){
		long crap=600851475143l;
		long alt=1;
		for(long i=1; i<crap; i++){
			if(crap%i==0){
				alt*=i;
				System.out.print(i+" ");
			}
			if(alt==crap){
				break;
			}
		}
	}
}

File: python/p003.py
def isPrime(num):
    if (num < 2):
        return False
    if (num == 2):
        return True
    if (num % 2 == 0):
        return False
    i = 3
    while (i * i <= num):
        if (num % i == 0):
            return False
        i += 2
    return True


number = 600851475143
x = int(number**0.5)
i = 3

while (i < x):
    if (number % i == 0 and isPrime(i)):
        print(i)
    i += 2
